clippy_lints v0.0.184 doesn't compile

I tried to add Clippy to my project today, but it didn't compile. I tried in Windows 10 and WSL (Ubuntu).
Rust nightly 1.25.0.

It's like the `label` field was removed from a struct and downstream code wasn't updated.


Error message

```
mason@mason-gaming-pc:~/dev/rust/golden$ cargo +nightly build --features "clippy" --verbose
       Fresh left-pad v1.0.1
       Fresh dtoa v0.4.2
       Fresh quote v0.3.15
       Fresh matches v0.1.6
       Fresh either v1.4.0
       Fresh lazy_static v1.0.0
       Fresh quine-mc_cluskey v0.2.4
       Fresh bitflags v0.9.1
       Fresh regex-syntax v0.4.2
       Fresh percent-encoding v1.0.1
       Fresh itoa v0.3.4
       Fresh libc v0.2.36
       Fresh vec_map v0.8.0
       Fresh unicode-normalization v0.1.5
       Fresh if_chain v0.1.2
       Fresh getopts v0.2.17
       Fresh utf8-ranges v1.0.0
       Fresh unicode-xid v0.0.4
       Fresh num-traits v0.1.42
       Fresh semver-parser v0.7.0
       Fresh lazy_static v0.1.16
       Fresh void v1.0.2
       Fresh serde v1.0.27
       Fresh unicode-bidi v0.3.4
       Fresh itertools v0.6.5
       Fresh memchr v2.0.1
       Fresh synom v0.11.3
       Fresh semver v0.6.0
       Fresh unreachable v1.0.0
       Fresh toml v0.4.5
       Fresh serde_json v1.0.9
       Fresh idna v0.1.4
       Fresh aho-corasick v0.6.4
       Fresh syn v0.11.11
       Fresh thread_local v0.3.5
       Fresh pulldown-cmark v0.0.15
       Fresh url v1.6.0
       Fresh serde_derive_internals v0.19.0
       Fresh regex v0.2.5
       Fresh serde_derive v1.0.27
   Compiling clippy_lints v0.0.184
       Fresh cargo_metadata v0.2.3
     Running `rustc --crate-name clippy_lints /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=cb8c0529306695b6 -C extra-filename=-cb8c0529306695b6 --out-dir /mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps -L dependency=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps --extern semver=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libsemver-d67c3d0a5514e050.rlib --extern pulldown_cmark=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libpulldown_cmark-b6274c104a4ed161.rlib --extern quine_mc_cluskey=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libquine_mc_cluskey-ebc359e1ace7a017.rlib --extern itertools=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libitertools-78ea1062c84dabe4.rlib --extern regex_syntax=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libregex_syntax-fa8602255742fda4.rlib --extern lazy_static=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/liblazy_static-81d07c4e79860199.rlib --extern if_chain=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libif_chain-5deb8c66f3e01bac.rlib --extern serde=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libserde-7017de094d273885.rlib --extern toml=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libtoml-2949ae399a9a9d88.rlib --extern unicode_normalization=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libunicode_normalization-9b8361452867f8a7.rlib --extern url=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/liburl-d537e41b61bb3fd7.rlib --extern serde_derive=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libserde_derive-0e212090d3b1aec7.so --extern matches=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libmatches-32edb66e6ed63589.rlib --cap-lints allow`
error[E0609]: no field `label` on type `rustc::hir::Destination`
  --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/hir_utils.rs:76:26
   |
76 |                 both(&li.label, &ri.label, |l, r| l.name.as_str() == r.name.as_str())
   |                          ^^^^^ unknown field
   |
   = note: available fields are: `ident`, `target_id`

error[E0609]: no field `label` on type `rustc::hir::Destination`
  --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/hir_utils.rs:76:37
   |
76 |                 both(&li.label, &ri.label, |l, r| l.name.as_str() == r.name.as_str())
   |                                     ^^^^^ unknown field
   |
   = note: available fields are: `ident`, `target_id`

error[E0609]: no field `name` on type `&_`
  --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/hir_utils.rs:76:53
   |
76 |                 both(&li.label, &ri.label, |l, r| l.name.as_str() == r.name.as_str())
   |                                                     ^^^^

error[E0609]: no field `label` on type `rustc::hir::Destination`
  --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/hir_utils.rs:90:26
   |
90 |                 both(&li.label, &ri.label, |l, r| l.name.as_str() == r.name.as_str())
   |                          ^^^^^ unknown field
   |
   = note: available fields are: `ident`, `target_id`

error[E0609]: no field `label` on type `rustc::hir::Destination`
  --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/hir_utils.rs:90:37
   |
90 |                 both(&li.label, &ri.label, |l, r| l.name.as_str() == r.name.as_str())
   |                                     ^^^^^ unknown field
   |
   = note: available fields are: `ident`, `target_id`

error[E0609]: no field `name` on type `&_`
  --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/hir_utils.rs:90:53
   |
90 |                 both(&li.label, &ri.label, |l, r| l.name.as_str() == r.name.as_str())
   |                                                     ^^^^

error[E0609]: no field `name` on type `&syntax::codemap::Spanned<syntax::ast::Symbol>`
   --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/hir_utils.rs:108:78
    |
108 |                 lls == rls && self.eq_block(lb, rb) && both(ll, rl, |l, r| l.name.as_str() == r.name.as_str())
    |                                                                              ^^^^

error[E0609]: no field `name` on type `&syntax::codemap::Spanned<syntax::ast::Symbol>`
   --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/hir_utils.rs:108:97
    |
108 |                 lls == rls && self.eq_block(lb, rb) && both(ll, rl, |l, r| l.name.as_str() == r.name.as_str())
    |                                                                                                 ^^^^

error[E0609]: no field `name` on type `&syntax::codemap::Spanned<syntax::ast::Symbol>`
   --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/hir_utils.rs:134:88
    |
134 |                 self.eq_expr(lc, rc) && self.eq_block(lb, rb) && both(ll, rl, |l, r| l.name.as_str() == r.name.as_str())
    |                                                                                        ^^^^

error[E0609]: no field `name` on type `&syntax::codemap::Spanned<syntax::ast::Symbol>`
   --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/hir_utils.rs:134:107
    |
134 |                 self.eq_expr(lc, rc) && self.eq_block(lb, rb) && both(ll, rl, |l, r| l.name.as_str() == r.name.as_str())
    |                                                                                                           ^^^^

error[E0609]: no field `label` on type `rustc::hir::Destination`
   --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/hir_utils.rs:330:36
    |
330 |                 if let Some(i) = i.label {
    |                                    ^^^^^ unknown field
    |
    = note: available fields are: `ident`, `target_id`

error[E0609]: no field `label` on type `rustc::hir::Destination`
   --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/hir_utils.rs:367:36
    |
367 |                 if let Some(i) = i.label {
    |                                    ^^^^^ unknown field
    |
    = note: available fields are: `ident`, `target_id`

error[E0609]: no field `name` on type `syntax::codemap::Spanned<syntax::ast::Symbol>`
   --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/hir_utils.rs:432:39
    |
432 |                     self.hash_name(&i.name);
    |                                       ^^^^ unknown field
    |
    = note: available fields are: `node`, `span`

error[E0609]: no field `name` on type `syntax::codemap::Spanned<syntax::ast::Symbol>`
   --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/hir_utils.rs:527:39
    |
527 |                     self.hash_name(&l.name);
    |                                       ^^^^ unknown field
    |
    = note: available fields are: `node`, `span`

error[E0599]: no method named `next_point` found for type `&syntax::codemap::CodeMap` in the current scope
   --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/sugg.rs:505:38
    |
505 |         let hi = cx.sess().codemap().next_point(remove_span).hi();
    |                                      ^^^^^^^^^^

error[E0609]: no field `label` on type `rustc::hir::Destination`
    --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/loops.rs:1811:50
     |
1811 |         ExprBreak(dest, ref passed_expr) if dest.label.is_none() && passed_expr.is_none() => true,
     |                                                  ^^^^^ unknown field
     |
     = note: available fields are: `ident`, `target_id`

error[E0599]: no method named `next_point` found for type `&syntax::codemap::CodeMap` in the current scope
    --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/methods.rs:1170:54
     |
1170 |                 let next_point = cx.sess().codemap().next_point(fold_args[0].span);
     |                                                      ^^^^^^^^^^

error[E0061]: this function takes 2 parameters but 1 parameter was supplied
   --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/misc.rs:548:10
    |
548 |         .builtin_deref(true)
    |          ^^^^^^^^^^^^^ expected 2 parameters

error[E0061]: this function takes 2 parameters but 1 parameter was supplied
   --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/misc.rs:552:10
    |
552 |         .builtin_deref(true)
    |          ^^^^^^^^^^^^^ expected 2 parameters

error[E0023]: this pattern has 5 fields, but the corresponding tuple variant has 4 fields
   --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/misc_early.rs:299:24
    |
299 |                 if let ExprKind::Closure(_, _, ref decl, ref block, _) = closure.node {
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 4 fields, found 5

error[E0023]: this pattern has 5 fields, but the corresponding tuple variant has 4 fields
   --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/misc_early.rs:330:24
    |
330 |                 if let ExprKind::Closure(_, _, _, _, _) = t.node;
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 4 fields, found 5

error[E0609]: no field `label` on type `rustc::hir::Destination`
  --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/unused_label.rs:72:110
   |
72 |             hir::ExprBreak(destination, _) | hir::ExprAgain(destination) => if let Some(label) = destination.label {
   |                                                                                                              ^^^^^ unknown field
   |
   = note: available fields are: `ident`, `target_id`

error[E0609]: no field `name` on type `syntax::codemap::Spanned<syntax::ast::Symbol>`
  --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/unused_label.rs:76:42
   |
76 |                 self.labels.insert(label.name.as_str(), expr.span);
   |                                          ^^^^ unknown field
   |
   = note: available fields are: `node`, `span`

error: aborting due to 23 previous errors

error: Could not compile `clippy_lints`.

Caused by:
  process didn't exit successfully: `rustc --crate-name clippy_lints /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=cb8c0529306695b6 -C extra-filename=-cb8c0529306695b6 --out-dir /mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps -L dependency=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps --extern semver=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libsemver-d67c3d0a5514e050.rlib --extern pulldown_cmark=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libpulldown_cmark-b6274c104a4ed161.rlib --extern quine_mc_cluskey=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libquine_mc_cluskey-ebc359e1ace7a017.rlib --extern itertools=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libitertools-78ea1062c84dabe4.rlib --extern regex_syntax=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libregex_syntax-fa8602255742fda4.rlib --extern lazy_static=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/liblazy_static-81d07c4e79860199.rlib --extern if_chain=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libif_chain-5deb8c66f3e01bac.rlib --extern serde=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libserde-7017de094d273885.rlib --extern toml=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libtoml-2949ae399a9a9d88.rlib --extern unicode_normalization=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libunicode_normalization-9b8361452867f8a7.rlib --extern url=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/liburl-d537e41b61bb3fd7.rlib --extern serde_derive=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libserde_derive-0e212090d3b1aec7.so --extern matches=/mnt/c/Users/mason_000/wsl/dev/rust/golden/target/debug/deps/libmatches-32edb66e6ed63589.rlib --cap-lints allow` (exit code: 101)
mason@mason-gaming-pc:~/dev/rust/golden$
```

```
mason@mason-gaming-pc:~/dev/rust/golden$ rustup show
Default host: x86_64-unknown-linux-gnu

installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu
nightly-x86_64-unknown-linux-gnu (default)

active toolchain
----------------

nightly-x86_64-unknown-linux-gnu (default)
rustc 1.25.0-nightly (73ac5d6a8 2018-01-11)
```

```
PS C:\Users\mason_000\wsl\dev\rust\advent2017> rustup show
Default host: x86_64-pc-windows-msvc

installed toolchains
--------------------

stable-x86_64-pc-windows-msvc
beta-x86_64-pc-windows-msvc
nightly-x86_64-pc-windows-msvc (default)

active toolchain
----------------

nightly-x86_64-pc-windows-msvc (default)
rustc 1.25.0-nightly (73ac5d6a8 2018-01-11)
```

I updated my toolchain and got a different error:

```
error[E0063]: missing field `approximate` in initializer of `rustc_errors::CodeSuggestion`
   --> /mnt/c/Users/mason_000/wsl/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.184/src/utils/mod.rs:636:16
    |
636 |     let sugg = rustc_errors::CodeSuggestion {
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `approximate`

error: aborting due to previous error

error: Could not compile `clippy_lints`.
```
```
mason@mason-gaming-pc:~/dev/rust/golden$ rustup show
Default host: x86_64-unknown-linux-gnu

installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu
nightly-x86_64-unknown-linux-gnu (default)

active toolchain
----------------

nightly-x86_64-unknown-linux-gnu (default)
rustc 1.25.0-nightly (616b66dca 2018-02-02)
```

*clippy 0.0.185*

error[E0599]: no method named `layout_of` found for type `(rustc::ty::TyCtxt<'a, 'tcx, 'tcx>, rustc::ty::ParamEnv<'tcx>)` in the current scope
    --> /Users/chentao/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.185/src/utils/mod.rs:1041:28
     |
1041 |     (cx.tcx, cx.param_env).layout_of(ty)
     |                            ^^^^^^^^^

error[E0599]: no method named `layout_of` found for type `(rustc::ty::TyCtxt<'a, 'tcx, 'tcx>, rustc::ty::ParamEnv<'tcx>)` in the current scope
    --> /Users/chentao/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.185/src/utils/mod.rs:1063:28
     |
1063 |     (cx.tcx, cx.param_env).layout_of(ty).ok().map(|layout| layout.align)
     |                            ^^^^^^^^^

error: aborting due to 2 previous errors

error: Could not compile `clippy_lints`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `clippy v0.0.185`, intermediate artifacts can be found at `/var/folders/ct/schmpgyd58b3r1dtr7nxdpvh0000gn/T/cargo-install.ujwvod8pGg0w`

Caused by:
  build failed

--- a/clippy_lints/src/utils/mod.rs
+++ b/clippy_lints/src/utils/mod.rs
@@ -9,7 +9,7 @@ use rustc::lint::{LateContext, Level, Lint, LintContext};
 use rustc::session::Session;
 use rustc::traits;
 use rustc::ty::{self, Ty, TyCtxt};
-use rustc::ty::layout::{LayoutOf, Align};
+use rustc::ty::layout::Align;
 use rustc_errors;
 use std::borrow::Cow;
 use std::env;
@@ -1038,7 +1038,7 @@ pub fn is_try(expr: &Expr) -> Option<&Expr> {
 }
 
 pub fn type_size<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, ty: Ty<'tcx>) -> Option<u64> {
-    (cx.tcx, cx.param_env).layout_of(ty)
+    cx.tcx.layout_of(cx.param_env.and(ty))
         .ok()
         .map(|layout| layout.size.bytes())
 }
@@ -1060,5 +1060,7 @@ pub fn get_arg_name(pat: &Pat) -> Option<ast::Name> {
 
 /// Returns alignment for a type, or None if alignment is undefined
 pub fn alignment<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, ty: Ty<'tcx>) -> Option<Align> {
-    (cx.tcx, cx.param_env).layout_of(ty).ok().map(|layout| layout.align)
+    cx.tcx.layout_of(cx.param_env.and(ty))
+        .ok()
+        .map(|layout| layout.align)
 }

